spec.md

```diff
@@ -5,4599 +5,421 @@ keywords: registry, on-prem, images, tags, repository, distribution, api, advanc
-
-###### On Success: OK
-
-```HTTP
-200 OK
+Location: <location>
 ```
 
-The API implements V2 protocol and is accessible.
+The `<location>` is a pullable manifest URL.
 
-###### On Failure: Not Found
+An attempt to pull a nonexistent repository MUST return response code `404 Not Found`
 
-```HTTP
-404 Not Found
-```
+#### Content Discovery
 
-The registry does not implement the V2 API.
+Currently, the only functionality provided by this workflow is the ability to discover tags.
 
-###### On Failure: Authentication Required
+To fetch the list of tags, perform a `GET` request to a path in the following format:
+[8a](#Endpoints) `/v2/<name>/tags/list`
 
-```HTTP
-401 Unauthorized
-WWW-Authenticate: <scheme> realm="<realm>", ..."
-Content-Length: <length>
-Content-Type: application/json
+`<name>` is the namespace of the repository. Assuming a repository is found, this request MUST return a
+`200 OK` response code. The list of tags MAY be empty, if there are no tags on the repository. If the list is not empty,
+the tags MUST be in lexical order (i.e. case-insensitive alphanumeric order).
 
+Upon success, the response MUST be a json body in the following format:
+```json
 {
-    "errors": [
-        {
-            "code": "<error code>",
-            "message": "<error message>",
-            "detail": ...
-        },
-        ...
-    ]
+  "name": "<name>",
+  "tags": [
+    "<tag1>",
+    "<tag2>",
+    "<tag3>"
+  ]
 }
 ```
 
-The client is not authenticated.
+`<name>` is the namespace of the repository, and `<tag1>`, `<tag2>`, and `<tag3>` are each tags on the repository.
 
-The following headers will be returned on the response:
+In addition to fetching the whole list of tags, a subset of the tags can be fetched by providing the `n` query parameter.
+In this case, the path will look like the following:
+[8b](#Endpoints) `/v2/<name>/tags/list?n=<int>`
 
-| Name               | Description                                           |
-|--------------------|-------------------------------------------------------|
-| `WWW-Authenticate` | An RFC7235 compliant authentication challenge header. |
-| `Content-Length`   | Length of the JSON response body.                     |
+`<name>` is the namespace of the repository, and `<int>` is an integer specifying the number of tags requested. The response
+to such a request MAY return fewer than `<int>` results, but only when the total number of tags attached to the repository
+is less than `<int>`. Otherwise, the response MUST include `<int>` results. Without the `last` query parameter (described
+next), the list returned will start at the beginning of the list and include `<int>` results. As above, the tags MUST be
+in lexical order.
 
-The error codes that MAY be included in the response body are enumerated below:
+The `last` query parameter provides further means for limiting the number of tags. It is used exclusively in combination with the
```

Comment:
'last' seems an odd name.  Wouldn't something like 'startafter' be more correct?

Reply:
This is how the prior spec was written, and how all registries currently operate. I agree that 'startafter' might be more correct, but the proposed v1 spec is more descriptive than prescriptive, and we don't want the spec to be out of sync with how all registries currently operate.